docs/security.md

```diff
@@ -0,0 +1,194 @@
+# Security
+
+The OpenTelemetry Collector defaults to operating in a secure manner, but is
+configuration driven. This document captures important security aspects and
+considerations for the Collector. This document is intended for both end-users
```

Comment:
A follow-up task could be to add this to the website, or as a blog post. Having a single doc for both end-users and component developers might be confusing to end-users.

Reply:
Agreed -- the immediate goal is to close out the AWS security issues filed in contrib, but I plan to take a full pass on collector documentation and move stuff like this to the website.